src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/transformers/CoroutineSupportTransformers.kt

```diff
@@ -41,4 +44,50 @@ internal class CoroutineCancellabilitySupportTransformer(
         super.visitMethodInsn(opcode, owner, name, desc, itf)
     }
 
+}
+
+/**
+ * [CoroutineDelaySupportTransformer] substitutes each invocation of [kotlinx.coroutines.delay] function with
+ * zero delay time. For example, it will transform `delay(100)` to `delay(0)`.
+ *
+ * It is sufficient to only transform `delay(Long)` overload, because
+ * the `delay(Duration)` overload is compiled to the invocation of the first one.
+ */
+internal class CoroutineDelaySupportTransformer(
+    fileName: String,
+    className: String,
+    methodName: String,
+    adapter: GeneratorAdapter,
+) : ManagedStrategyMethodVisitor(fileName, className, methodName, adapter) {
+
+    override fun visitMethodInsn(opcode: Int, owner: String, name: String, desc: String, itf: Boolean) = adapter.run {
+        if (!isCoroutineDelay(owner, name, desc)) {
+            visitMethodInsn(opcode, owner, name, desc, itf)
+            return
+        }
+
+        Logger.warn { "Instrument delay function" }
```

Comment:
What is this logging useful for?

Several questions about our approach to logging in `transformation` classes in general.
If we add logging here, then the question is, in general, do we want to log all applications of different transformers? E.g., would it then make sense to log all instrumented field accesses in `SharedMemoryTransformer`? Would it be useful to log this? Wouldn't the log then become too large and verbose?

Reply:
This exact one I just forgot to remove.

Logging might be more useful in the strategies method, such as what handlers are called: beforeWrite/Read, beforeMethodCall, etc. But that would definitely be too much. Another option is to log the interleaving tree and the positions where the execution is right now with the corresponding switch locations taken in the code. But I don't know how to print the interleaving tree compactly.

Reply:
removed